src/metabase/upload.clj

```diff
@@ -247,10 +248,17 @@
   [driver db]
   (driver.u/supports? driver :upload-with-auto-pk db))
 
+(defn- unique-alias-fn [driver]
+  (let [max-length (max-column-bytes driver)]
+    (fn [base suffix]
+      (as-> (str base "_" suffix) %
+            (driver/escape-alias driver %)
+            (lib.util/truncate-alias % max-length)))))
```

Comment:
wdyt about pulling this out of `lib.util`? The fact it's in a `metabase.lib.util` namespace might have the QP team thinking they can just modify it without breaking CSV upload behaviour. I've started doing that in https://github.com/metabase/metabase/pull/44862. To minimize conflicts maybe it can wait until both these PRs have landed though

Reply:
Yeah I like that direction a lot. The way this uses legacy and current lib functions together also seems odd - perhaps we can find a shared home for them.

Definitely let's wait for dust to settle a bit before we do the move.